lib/src/model/home_route_model.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+
+enum HomeRouteAspects {
+  homeRouteNavigationAspect,
+  homeRouteTileAspect,
+}
+
+class HomeRouteModel extends InheritedModel<HomeRouteAspects> {
+  final int? homeRouteNavigationAspect;
+  final int? homeRouteTileAspect;
+
+  const HomeRouteModel({
+    super.key,
+    this.homeRouteNavigationAspect,
+    this.homeRouteTileAspect,
+    required super.child,
+  });
+
+  // Home Route Model Update Notifier
+  @override
+  bool updateShouldNotify(HomeRouteModel oldWidget) {
+    return homeRouteNavigationAspect != oldWidget.homeRouteNavigationAspect ||
+        homeRouteTileAspect != oldWidget.homeRouteTileAspect;
+  }
+
+  // Home Route Model Dependent Update Notifier
+  @override
+  bool updateShouldNotifyDependent(
+      HomeRouteModel oldWidget, Set<HomeRouteAspects> dependencies) {
+    if (homeRouteNavigationAspect != oldWidget.homeRouteNavigationAspect &&
```

Comment:
This is not a correct implementation of InheritedModel for several reasons:
1. You don't need the aspect vars on lines 9 & 10
2. The model doesn't contain any actual data. Instead of the aspect vars you need to add other variables that describe the model

Check the guide and example here: https://api.flutter.dev/flutter/widgets/InheritedModel-class.html

Reply:
The data is the navigation drawer selected tile and browser launched status. When that data changes, the widget rebuilds. It is the correct implementation. Run the app and see for yourself.

Reply:
Sorry but it's definitely not correct for the reasons I stated above. Even if it works, it's not correct. 
Besides - we don't need to keep browser launched status. For selected tile - why would we want to keep that - tapping on a menu item opens a new screen and when you go be it should not be selected anymore.

Reply:
You misunderstand. The navigation drawer tiles are numbered. When you select a new tile, the number changes. The browser launcher is a boolean, when the browser is launched, it changes to true. The aspect variables exist because the inherited model wraps a child widget. That child can be any widget. It wraps the "Our Site" tile in the navigation drawer for the browser launcher and it wraps the drawer itself.

Reply:
Pretty sure I don't misunderstand. Please take a good look at the example on this page: https://api.flutter.dev/flutter/widgets/InheritedModel-class.html
As you can see `LogoModel` contains some actual model data:
```
final Color? backgroundColor;
final bool? large;
```

In addition to that there is the aspect enum
`enum LogoAspect { backgroundColor, large }`
which is used to determine what aspect needs to be rebuilt depending on what model data changed.

This is purely on the technical side.
On the business logic side: 
1. We don't need to keep track of 'selected' tiles as this has no impact on the UI. We just react to the click event.
2. Similarly - launching the site doesn't need any tracking

Reply:
Look Here:
`BrowserViewModel(
              browserLaunchedAspect: _browserLaunched,
              child: ListTile(
                enableFeedback: true,
                leading: const Icon(Icons.link),
                onTap: () => {
                  setState(() {
                    _homeRouteTile = 8;
                    _browserLaunched = _launchBrowser(launchUrl);
                  }),
                },
                selected: _homeRouteTile == 8,
                title: const Text(
                  'Our Site',
                  softWrap: true,
                ),
              ),
            ),`

Look Here: 
`return HomeRouteModel(
      homeRouteTileAspect: _homeRouteTile,
      child: Drawer(`

Data:
`enum BrowserViewAspects {
  browserLaunchedAspect,
}

class BrowserViewModel extends InheritedModel<BrowserViewAspects> {
  final Future<void>? browserLaunchedAspect;

  const BrowserViewModel({
    super.key,
    this.browserLaunchedAspect,
    required super.child,
  });`

Data:
`enum HomeRouteAspects {
  homeRouteTileAspect,
}

class HomeRouteModel extends InheritedModel<HomeRouteAspects> {
  final int? homeRouteTileAspect;

  const HomeRouteModel({
    super.key,
    this.homeRouteTileAspect,
    required super.child,
  });`